jni/src/global_jni_env.c

```diff
@@ -0,0 +1,1188 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+#include "global_jni_env.h"
+jint Jei_GetVersion() {
+    attach_thread();
+    return (*jniEnv)->GetVersion(jniEnv);
```

Comment:
What is the conceptual relationship between dartjni.c and global_jni_env.c ?

jniEnv lives in dartjni.c while globalEnv abstracts over that.

Does the generated code talk with both? If yes, then it is not a full abstraction and maybe these two files should just be merged.

Reply:
JNIEnv lives in jni.h from NDK actually. `dartjni.c` provides other core infrastructure specific to dart. global_jni_env is just a wrapper over env with handful of methods.

True. The orgnization is clumsy and I should revisit it.